frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseOverrideForkPointAction.java

```diff
@@ -0,0 +1,144 @@
+    return getString("action.GitMachete.BaseOverrideForkPointAction.description-action-name");
+  }
+
+  @Override
+  public @I18nFormat({GENERAL, GENERAL}) String getEnabledDescriptionFormat() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.description");
+  }
+
+  @Override
+  public String getCurrentBranchText() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.text.current-branch");
+  }
+
+  @Override
+  public List<SyncToParentStatus> getEligibleStatuses() {
+    return List.of(SyncToParentStatus.InSyncButForkPointOff);
+  }
+
+  @Override
+  @UIEffect
+  public void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+    syncToParentStatusDependentActionUpdate(anActionEvent);
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    var project = getProject(anActionEvent);
+    var selectedVcsRepository = getSelectedGitRepository(anActionEvent).getOrNull();
+    var branchUnderAction = getNameOfBranchUnderAction(anActionEvent);
+    var branch = branchUnderAction.flatMap(pn -> getGitMacheteBranchByName(anActionEvent, pn)).getOrNull();
+
+    if (selectedVcsRepository == null || branch == null || branch.isRootBranch()) {
+      return;
+    }
+
+    var nonRootBranch = branch.asNonRootBranch();
+    var selectedCommit = new OverrideForkPointDialog(project, nonRootBranch.getParentBranch(), nonRootBranch)
+        .showAndGetSelectedCommit();
+    if (selectedCommit == null) {
+      log().debug(
+          "Selected commit to be the new fork point is null: most likely the action has been canceled from override-fork-point dialog");
+      return;
+    }
+
+    LOG.debug("Enqueueing fork point override");
+    new Task.Backgroundable(project, "Overriding fork point...") {
+      @Override
+      public void run(ProgressIndicator indicator) {
+        overrideForkPoint(anActionEvent, branch, selectedCommit);
+      }
+    }.queue();
+  }
+
+  /**
+   * This method must NOT be called on UI thread.
+   */
+  private void overrideForkPoint(AnActionEvent anActionEvent, IGitMacheteBranch branch, IGitMacheteCommit forkPoint) {
+    var selectedVcsRepository = getSelectedGitRepository(anActionEvent);
+
+    if (selectedVcsRepository.isDefined()) {
+      var root = selectedVcsRepository.get().getRoot();
+      var project = getProject(anActionEvent);
+      setOverrideForkPointConfigKeyValues(project, root, branch.getName(), forkPoint, branch.getPointedCommit());
+    }
+
+    getGraphTable(anActionEvent).queueRepositoryUpdateAndModelRefresh();
+  }
+
+  /**
+   * This method must NOT be called on UI thread.
+   */
+  private void setOverrideForkPointConfigKeyValues(
+      Project project,
+      VirtualFile root,
+      String branchName,
+      IGitMacheteCommit forkPoint,
+      IGitMacheteCommit ancestorCommit) {
+    var section = "machete";
+    var subsectionPrefix = "overrideForkPoint";
```

Comment:
Whoops while you're here... com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java:487:
```java
// Name spans the characters after the first dot
```
this should obviously be:
```java
// Name spans the characters after the last dot
```